hazelcast-sql/src/main/java/com/hazelcast/sql/impl/schema/Mapping.java

```diff
@@ -117,7 +134,11 @@ public void writeData(ObjectDataOutput out) throws IOException {
     @Override
     public void readData(ObjectDataInput in) throws IOException {
         name = in.readString();
-        externalName = in.readString();
+        if (in.getVersion().isGreaterOrEqual(Versions.V5_3)) {
+            externalName = in.readStringArray();
+        } else {
+            externalName = new String[]{in.readString()};
```

Comment:
what if the name from 5.2 contains dots?

Reply:
It should be identical to the case where you quote it in 5.3 `CREATE MAPPING .... EXTERNAL NAME \"map.with.dot\"`

Reply:
In 5.2 we didn't support tables in non-default schema in Jdbc connector.

Moreover the Jdbc connector is not backward compatible because of the external-data-store -> data link change (it's allowed because it was released as beta).